sympde/expr/evaluation.py

```
@@ -434,218 +493,239 @@ def __getitem__(self, indices, **kw_args):
+            return cls.eval(expr.arg, domain=domain).inv()
 
         elif isinstance(expr, ScalarFunction):
             return expr
 
         elif isinstance(expr, VectorFunction):
             return ImmutableDenseMatrix([[expr[i]] for i in range(dim)])
 
+        elif isinstance(expr, (minus, plus)):
+            newexpr = cls.eval(expr.args[0], domain=domain)
+            return type(expr)(newexpr)
+
         elif isinstance(expr, PullBack):
-            return cls.eval(expr.expr, dim=dim, logical=True)
+            return cls.eval(expr.expr, domain=domain)
 
         elif isinstance(expr, MatrixElement):
-            base = cls.eval(expr.base, dim=dim, logical=logical)
+            base = cls.eval(expr.base, domain=domain)
             return base[expr.indices]
 
         elif isinstance(expr, BasicForm):
             # ...
             dim     = expr.ldim
-            domain  = expr.domain
+            domains  = expr.domain
 
-            if not isinstance(domain, Union):
-                logical = domain.mapping is None
-                domain  = (domain,)
+            if not isinstance(domains, Union):
+                domains  = (domains,)
             # ...
             d_expr = OrderedDict()
-            for d in domain:
+            d_int  = OrderedDict()
+            for d in domains:
                 d_expr[d] = S.Zero
             # ...
             if isinstance(expr.expr, Add):
                 for a in expr.expr.args:
-                    newexpr = cls.eval(a, dim=dim, logical=logical)
 
                     # ...
                     try:
-                        domain = _get_domain(a)
-                        if not isinstance(domain, Union):
-                            domain = (domain, )
+                        domains = _get_domain(a)
+                        if not isinstance(domains, Union):
+                            domains = (domains, )
                     except:
                         pass
```

Ehm... why are we ignoring any exception raised by `_get_domain(a)`?

---

we should maybe delete the try-except clause

---

Yes please!

---

done